docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,513 @@
+
+MongoDB with Panache provides active record style entities (and repositories) like you have in Hibernate ORM with Panache and focuses on making your entities trivial and fun to write in Quarkus.
+
+== First: an example
+
+What we're doing in Panache is allow you to write your MongoDB entities like this:
+
+[source,java]
+--
+public class Person extends PanacheMongoEntity {
+    public String name;
+    public LocalDate birth;
+    public Status status;
+
+    public static Person findByName(String name){
+      return find("name", name).firstResult();
+    }
+
+    public static List<Person> findAlive(){
+      return list("status", Status.Alive);
+    }
+
+    public static void deleteStefs(){
+      delete("name", "Stef");
+    }
+}
+--
+
+You have noticed how much more compact and readable the code is?
+Does this look interesting? Read on!
+
+NOTE: the `list()` method might be surprising at first. It takes fragments of PanacheQL queries (subset of JPQL) and contextualize the rest.
+That makes for very concise but yet readable code.
+MongoDB native queries are also supported.
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `using-mongodb-panache` {quickstarts-tree-url}/using-mongodb-panache[directory].
+
+== Creating the Maven project
+
+First, we need a new project. Create a new project with the following command:
+
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=using-mongodb-panache \
+    -DclassName="org.acme.rest.json.FruitResource" \
+    -Dpath="/fruits" \
+    -Dextensions="resteasy-jsonb,mongodb-panache"
+----
+
+This command generates a Maven structure importing the RESTEasy/JAX-RS, JSON-B and MongoDB with Panache extensions.
+After this, the quarkus-mongodb-panache extension has been added to your `pom.xml`.
+
+If you don't want to generate a new project you can add the dependency in your `pom.xml`:
+
+[source,xml]
+--
+<dependencies>
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-mongodb-panache</artifactId>
+    </dependency>
+</dependencies>
+--
+
+== Setting up and configuring MongoDB with Panache
+
+To get started:
+
+* add your settings in `{config-file}`
+* Make your entities extend `PanacheMongoEntity`, you can use the `@MongoEntity` annotation to specify the name of the database and the name of the collection (it will default to the name of your entity).
+* place your entity logic in static methods in your entities
+
+Follow the link:mongo-guide.html#configuring-the-mongodb-database[Configuring the MongoDB database].
```

Comment:
Do we really want people to go to this guide and follow it or should it only be given as a reference after the configuration?

Reply:
I move it at the end of the paragraph and transform the sentence as a reference link not a link to follow before continuing.